runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10338,6 +10338,317 @@ inlineCompareAndSwapNative(
+
+   // if loopLimit > 4, jump to fiveOrMoreBytesLabel
+   generateRegImmInstruction(TR::InstOpCode::CMP4RegImm4, node, loopLimitReg, 4, cg);
+   generateLabelInstruction(TR::InstOpCode::JG4, node, fiveOrMoreBytesLabel, cg);
+
+   // Load the first two bytes at address [buf + index] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::L2RegMem, node, chunkReg, generateX86MemoryReference(bufReg, indexReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg), cg);
+
+   // OR the second two bytes at address [buf + (limit - 2)] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::OR2RegMem, node, chunkReg, generateX86MemoryReference(bufReg, limitReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes() - 2, cg), cg);
+
+   // Jump to residualTestLabel
+   generateLabelInstruction(TR::InstOpCode::JMP4, node, residualTestLabel, cg);
+
+
+   // Case in which there are five or more residual bytes
+   generateLabelInstruction(TR::InstOpCode::label, node, fiveOrMoreBytesLabel, cg);
+
+   // Load the first four bytes at address [buf + index] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::L4RegMem, node, chunkReg, generateX86MemoryReference(bufReg, indexReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg), cg);
+
+   // OR the second four bytes at address [buf + (limit - 4)] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::OR4RegMem, node, chunkReg, generateX86MemoryReference(bufReg, limitReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes() - 4, cg), cg);
+
+   // Jump to residualTestLabel
+   generateLabelInstruction(TR::InstOpCode::JMP4, node, residualTestLabel, cg);
+
+
+   // Case in which there are nine or more residual bytes
+   generateLabelInstruction(TR::InstOpCode::label, node, nineOrMoreBytesLabel, cg);
+
+   // Load the first eight bytes at address [buf + index] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::L8RegMem, node, chunkReg, generateX86MemoryReference(bufReg, indexReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg), cg);
+
+   // OR the second eight bytes at address [buf + (limit - 8)] into the chunk register
+   generateRegMemInstruction(TR::InstOpCode::OR8RegMem, node, chunkReg, generateX86MemoryReference(bufReg, limitReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes() - 8, cg), cg);
+
+
+   // Examine the chunk register now that all of the residual bytes have been ORed into it
+   generateLabelInstruction(TR::InstOpCode::label, node, residualTestLabel, cg);
+
+   // AND the residual bytes with the new mask
+   generateRegRegInstruction(TR::InstOpCode::TEST8RegReg, node, chunkReg, maskReg, cg);
+
+   // If the result is nonzero (i.e. at least one of the sign bits is set), jump to returnHasNegativesLabel
+   generateLabelInstruction(TR::InstOpCode::JNE4, node, returnHasNegativesLabel, cg);
+
+
+   // returnNoNegatives label
```

Comment:
A general comment throughout this PR.  You shouldn't comment trivially obvious code, or use comments that essentially repeat what the code says equally or better (another example: "Jump to residualTestLabel").

The instruction sequence being laid out is already long and branchy, and fewer unnecessary comments to fill up a reader's context buffer improves readability immensely.

Reply:
I'm going to mark this comment as "Unresolved".  I just wanted @0xdaryl to have a chance to verify whether he's satisfied with the comments.